src/main/java/janggi/domain/Turn.java

```diff
@@ -0,0 +1,29 @@
+package janggi.domain;
+
+import janggi.domain.piece.Team;
+
+public class Turn {
+    private Team turn;
+
+    private Turn(final Team turn) {
+        this.turn = turn;
+    }
+
+    public static Turn startWith(final Team team) {
+        return new Turn(team);
+    }
+
+    public Team next() {
+        switchTeam(turn);
+        return turn;
+    }
+
+    private void switchTeam(final Team currentTurn) {
+        if (currentTurn == Team.RED) {
+            turn = Team.BLUE;
+        }
+        if (currentTurn == Team.BLUE) {
+            turn = Team.RED;
+        }
+    }
```

Comment:
currentTurn을 적어서 의미를 알 수 있어서 좋긴한데, 한편으로는 굳이 인스턴스 변수를 인자로 받아야하는지 고민이 되긴 하네요. 대니의 생각을 듣고 싶네요.

Reply:
작성 의도는 의미 전달이 맞습니다. 🙂 베루스 의견을 듣고 다시 생각해봤는데요, 메서드 명을 `switchTurn`으로 고치고, 인자를 받지 않으니 어느정도 의미 전달을 하면서 불필요한 인자를 없앨 수 있어서 좋은 것 같은데, 어떻게 생각하시나요?

```suggestion
    private void switchTurn() {
        if (turn == Team.RED) {
            turn = Team.BLUE;
        }
        if (turn == Team.BLUE) {
            turn = Team.RED;
        }
    }
```

Reply:
제안하신 방법도 좋은 것 같습니다! 구조적으로 중요치 않은 부분이라 편한 방법으로 구현하면 될 것 같습니다. 인스턴스 변수가 있음에도 사용하지 않은 것이 궁금해서 여쭤본거였습니다~